cfme/tests/infrastructure/test_tenant_quota.py

```diff
@@ -103,6 +104,39 @@ def small_vm(provider, small_template_modscope):
     vm.cleanup_on_provider()
 
 
+@pytest.fixture()
+def new_vm(setup_provider, provider):
+    """Fixture to provision appliance to the provider being tested if necessary"""
+    vm_name = random_vm_name(context='migrate')
+    try:
+        template_name = provider.data.templates.small_template.name
```

Comment:
Required: this check can be added to pytestmark
ex. pytest.mark.provider([VMwareProvider, RHEVMProvider], scope="module", required_fields=[['templates', 'small_template'] selector=ONE_PER_TYPE)